src/js/hooks/useCheckRightSpace.jsx

```diff
@@ -0,0 +1,9 @@
+const useCheckRightSpace = (elementReference, isVisible) => {
+  if (elementReference.current && isVisible) {
+    const elementCoordinates = elementReference.current.getBoundingClientRect();
+    return window.innerWidth > elementCoordinates.right + elementCoordinates.width;
+  }
+  return false;
+};
+
+export default useCheckRightSpace;
```

Comment:
this is not a hook - what I meant is to move `ref` and `states` logic to a hook, so you could return them in the hook like this:
```
return { dropdownRef, isOpen, setIsOpen }
```
and you could use it in the component like this:
```
const { dropdownRef, isOpen, setIsOpen } = useCheckRightSpace();
```
remember to change the names in the hook to generic ones, but in the component you could do:
```
const { dropdownRef: <ref_name_from_hook> } 
```

Reply:
remember also about moving your function `shouldAlignLeft` to the hook